### DISCUSSION TITLE
CORS/Cookies on Android under Capacitor 4.x

### DISCUSSION THREAD

**Discussion by rj33** [ionic-team/capacitor#6438] [Q&A]
I'm attempting to upgrade from Capacitor 3.x to Capacitor 4.

I have an app that targets both iOS and Android. 

I've just used the migrate command, and so far it looks like iOS is working fine.

Unfortunately I appear to be hitting what is likely to be either CORS and/or Cookie issues with Android.

I didn't want to file a bug report yet, as I suspect the issue is at my end, and I've got several years worth of hacks built up to deal with iOS and CORS/Cookie problems, but up until now Android has tended to work well out-of-the-box.

Unfortunately I can't make use of things like the Http plugin, as I don't believe it attempts to intercept websocket calls,  which I make extensive use of.

The server section of my config just has a 'hostname' entry, which for now was enough to get my particular setup working when communicating with the live server. However right now I can't get auth to work on Android (still working on iOS), and I wondered if anyone else had run into new issues around CORS/Cookies after upgrading to Capacitor 4? My best guess is that the higher API target used on Android now has brought in some new treatment for cookies/CORS that wasn't required previously, but it isn't obvious to me yet what that might be.

**Comment by rj33** [comment]
Actually just after posting this, I found that setting "androidSchema": "https" fixed my issue.

The origin is set to http://hostnamefromconfigfile by default, which was probably not matching my CORS specs on the server properly, I did have an entry that I thought was catching this, but perhaps latest android API versions are not as forgiving on the CORS hostname matches. In any case setting the schema to be https, set the origin to https://hostnamefromconfigfile and everything started working again as it had in capacitor 3.x. 

Is there any reason not to default the schema to https instead of http?

**Comment by ghost** [comment]
As per https://github.com/ionic-team/capacitor/pull/6342 it's already the default in capacitor 5.
